### DISCUSSION TITLE
GOES L1B serverside spatial subsetter?

### DISCUSSION THREAD

**Discussion by lukevalin** [blaylockbk/goes2go#15] [Q&A] [answered]
Hi Brian,

I'm wondering if you know of a straightforward way to be able to spatially subset GOES L1B on AWS before downloading?

--Luke

**Comment by blaylockbk** [comment, marked as answer]
Hi @lukevalin 
I'm not aware of any methods to subset the GOES NetCDF files from AWS. You aren't the first to ask. This topic is of interest to many, and I think there are some who are developing this capability.

**Comment by blaylockbk** [comment]
If you do a search for "Cloud-optimized NetCDF" you will learn that NetCDF files are not very efficient in the cloud for tasks like subsetting. Here are some articles to read:

* https://medium.com/pangeo/fake-it-until-you-make-it-reading-goes-netcdf4-data-on-aws-s3-as-zarr-for-rapid-data-access-61e33f8fe685
* https://medium.com/pangeo/cloud-performant-reading-of-netcdf4-hdf5-data-using-the-zarr-library-1a95c5c92314
* http://matthewrocklin.com/blog/work/2018/02/06/hdf-in-the-cloud

Some alternative formats are to use cloud-optimized Geotiff or Zarr. But I'm not aware that the GOES dataset is available in these formats.

**Comment by lukevalin** [comment]
Thanks Brian. The information you provided was very helpful.
